File: ImYourSouffleur.Server/Models/Request/ChatHistoryRequest.cs
namespace ImYourSouffleur.Server.Models.Request
{
    public class ChatHistoryRequest
    {
        public string Context { get; set; }
        public List<ChatMessage> Messages { get; set; } = new();
    }

    public class ChatMessage
    {
        public AuthorRole Role { get; set; }
        public string Content { get; set; } = string.Empty;
     
    }

    public enum AuthorRole
    {
        User,
        Assistant,
        System
    }
}

File: imyoursouffleur.client/src/models/ChatHistoryRequest.ts


export enum AuthorRole {
    User,
    Assistant,
    System
}

export class ChatHistoryRequest {
    messages: ChatMessage[];
    context: string;

    constructor(context:string, messages: ChatMessage[]) {       
        this.messages = messages;
        this.context = context;
    }
}

export class ChatMessage {
    role: AuthorRole;
    content: string;
    constructor(content: string, role: AuthorRole) {
        this.role = role;
        this.content = content;
    }
}